src/modules/tools/temperaturecontrol/TemperatureControl.cpp

```diff
@@ -327,15 +327,26 @@ void TemperatureControl::on_gcode_received(void *argument)
 
         } else if( ( gcode->m == this->set_m_code || gcode->m == this->set_and_wait_m_code ) && gcode->has_letter('S')) {
             // this only gets handled if it is not controlled by the tool manager or is active in the toolmanager
-            this->active = true;
-
-            // this is safe as old configs as well as single extruder configs the toolmanager will not be running so will return false
-            // this will also ignore anything that the tool manager is not controlling and return false, otherwise it returns the active tool
-            void *returned_data;
-            bool ok = PublicData::get_value( tool_manager_checksum, is_active_tool_checksum, this->name_checksum, &returned_data );
-            if (ok) {
-                uint16_t active_tool_name =  *static_cast<uint16_t *>(returned_data);
-                this->active = (active_tool_name == this->name_checksum);
+
+            if(!gcode->has_letter('T') || gcode->is_modal_t) { //Default temperature handling, just test if we are the active tool head
+                
+                // this is safe as old configs as well as single extruder configs the toolmanager will not be running so will return false
+                // this will also ignore anything that the tool manager is not controlling and return false, otherwise it returns the active tool
+           
+                this->active = true;
+
+                void *returned_data;
+                bool ok = PublicData::get_value( tool_manager_checksum, is_active_tool_checksum, this->name_checksum, &returned_data );
+                if (ok) {
+                    uint16_t active_tool_name =  *static_cast<uint16_t *>(returned_data);
+                    this->active = (active_tool_name == this->name_checksum);                
+                }
+            }
+            else if(gcode->has_letter('T') && this->designator[0]=='T'){ //Non-modal T method handling, if T was specified test if we match that designator, even if we aren't the active tool
+                if(this->designator.length()==2 && gcode->get_uint('T')<10) //Support M104 T[0-9] S210 type of call
+                    this->active = ((int)(this->designator[1]-'0') == gcode->get_int('T'));
+                else
+                    this->active = gcode->get_uint('T') == 0; //Used when the tool 0 temperature control identifier is just 'T', rather than 'T0'
```

Comment:
Just T is an illegal gcode, ALL gcode parameters MUST have a letter and a number this is a NIST standard.

Reply:
That is to catch the configuration specified identifier, not the gcode.
See default smoothie config example here where,  the first is labeled only "T" and the second tool is T1
https://github.com/Smoothieware/Smoothieware/blob/edge/ConfigSamples/Smoothieboard/config